### PR TITLE
Add `TunableMixin` and `TuneAnalysis`

### LINKED ISSUE #1799: Define `_tunables` class property at the base level
This would make it easier to expose model classes to `autotune.ModelTuner` as any model class inheriting from `BaseModelClass` would work properly out of the box.

### PR DESCRIPTION
**Pull request by martinkim0**
Closes #1799

### CODE REVIEW

File: scvi/autotune/_manager.py [hunk 1]
```diff
@@ -86,31 +88,58 @@ def _cls_to_tunable_type(cls: Any) -> str:
                     return tunable_type
             return "unknown"
 
+        def _parse_func_params(func: Callable, parent: Any, tunable_type: str) -> dict:
+            # parse function docstring for parameter descriptions
+            docstring = None
+            if func.__name__ == "__init__":
+                # __init__ func docstrings are the class's docstring
+                docstring = parse(parent.__doc__)
```
Comment (adamgayoso) [inline scvi/autotune/_manager.py:96]: I think we might be able to use docrep for this instead of introducing another dependency?

Comment (martinkim0) [reply scvi/autotune/_manager.py:96]: Might just remove the functionality to print out keyword argument descriptions, I think it's too overloaded with it.

File: scvi/train/_trainer.py [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
+from scvi.autotune._types import Tunable, TunableMixin
 
 from ._callbacks import LoudEarlyStopping
 from ._logger import SimpleLogger
 from ._progress import ProgressBar
 from ._trainingplans import PyroTrainingPlan
 
 
-class Trainer(pl.Trainer):
+class Trainer(pl.Trainer, TunableMixin):
```
Comment (adamgayoso) [inline scvi/train/_trainer.py:18]: why is the mixin going last here? This applies to a lot of them.

Comment (martinkim0) [reply scvi/train/_trainer.py:18]: Should it be going first? Wasn't sure if the order mattered

Comment (adamgayoso) [reply scvi/train/_trainer.py]: It does matter if there were collisions in method names (lookup MRO python), it's probably best to have the mixins before the base class that provides most functionality

**Comment by martinkim0** [comment]
@adamgayoso Any idea what could be happening with docs fail? Haven't been able to figure it out

**Comment by martinkim0** [comment]
Looks like all new docs builds are failing with the same error

File: .gitignore [hunk 3]
```diff
@@ -113,6 +113,7 @@ ENV/
 .floo
 .flooignore
 
+tests/data
```
Comment (adamgayoso) [inline .gitignore:116]: why is this in gitignore?

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (martinkim0) [reply .gitignore:116]: The autotune tests leave some log files there

Comment (martinkim0) [reply .gitignore:116]: I can change where they're logged

**Review by adamgayoso — APPROVED** [review]

**Pull request merged**